language/tools/mirai-dataflow-analysis/src/configuration.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{datalog::DatalogBackend, util::make_absolute};
+
+/// Specifies reduction operations that may be performed
+/// on a call graph. Supported operations are:
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum CallGraphReduction {
+    /// Only include nodes reachable from the given node.
+    /// See `CallGraph::filter_reachable`.
+    Slice(Box<str>),
+    /// Remove nodes in the graph that belong to crates other than
+    /// `CallGraphConfig.included_crates`. The outgoing edges of these
+    /// removed node are connected to the node's parents.
+    /// See `CallGraph::fold_excluded`.
+    Fold,
+    /// Remove duplicated edges (only considers edge endpoints).
+    /// See `CallGraph::deduplicate_edges`.
+    Deduplicate,
+    /// Remove nodes that have no incoming or outgoing edges.
+    /// See `CallGraph::filter_no_edges`.
+    Clean,
+}
+
+/// Configuration options for Datalog output
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DatalogConfig {
+    /// Specifies location for graph to be output as Datalog input relations.
+    pub ddlog_output_path: PathBuf,
```

Comment:
It's a carry-over from when only Differential Datalog was used (though it could also be viewed as a short form of "Datalog"). In any case, this part of the configuration is automatically generated so it shouldn't affect the end user.